src/Listener/PodeRequest.cs

```diff
@@ -221,42 +237,50 @@ private bool ValidateCertificateCallback(object sender, X509Certificate certific
         /// <returns>A Task representing the async operation, with a boolean indicating whether the connection should be closed.</returns>
         public async Task<bool> Receive(CancellationToken cancellationToken)
         {
-            // Check if the stream is open
-            if (State != PodeStreamState.Open)
-            {
-                return false;
-            }
-
+            await StreamLock.WaitAsync(cancellationToken);
```

Comment:
yes you are right. I added it when I was trying to fix the reported issue.

Reply:
I have to reinstate it because otherwise everything fails

Reply:
For this, I was more referring that the `StreamLock` variable entirely can go - because of how code will drop into this method, Semaphores shouldn't be required, and the locking will just add overhead.

Reply:
I cannot remove the Semaphores. there are multiple receive calls I presume.